packages/react-component-library/src/components/Modal/Footer.tsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react'
+import { Button, ButtonProps } from '../Button'
+
+export interface FooterProps {
+  primaryButton?: ButtonProps
+  secondaryButton?: ButtonProps
+  tertiaryButton?: ButtonProps
+}
+
+export const Footer: React.FC<FooterProps> = ({
+  primaryButton,
+  secondaryButton,
+  tertiaryButton,
+}) => {
+  return (
+    <footer className="rn-modal__footer" data-testid="footer">
```

Comment:
with test id's I'm starting to namespace them with the component name.. if a sub component has a data test id it can lead to false positives or picking the wrong element.

Reply:
Hmm fair enough - we should probably be shallow mounting most of the time and testing from atoms up, and having the tests stub out sub-components where possible.

Reply:
Does this tool allow shallow mount? I like how it works not tbh :)

Reply:
Hmm interestingly it doesn't support it:

https://kentcdodds.com/blog/why-i-never-use-shallow-rendering

Oh well scrap that then